src/components/interface/AudioPlay.jsx

```diff
@@ -1,26 +1,78 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+
 import { withStyles } from '@material-ui/core/styles';
 import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';
+
 import styles from '../styles';
+import Button from '../interface/Button';
+import * as firebase from 'firebase';
 
 class AudioPlayer extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      disabled: false
+      audioFile: undefined,
+      metadataLoaded: false
     };
   }
 
-  // TODO: handle audio playing file
+  loadAudio = () => {
+    this.pathRef.getDownloadURL().then((url) => {
+      fetch(url).then(() => {
+        this.setState({ audioFile: new Audio(url) });
+        this.state.audioFile.addEventListener('loadeddata', () => {
+          this.setState({ metadataLoaded: true });
+        })
+      })
+    }).catch(function () {
+      console.log("ERROR. Firebase audio file failed to load.");
+    });;
+  }
+
+  componentDidMount() {
+    const { audioURL } = this.props;
+    const { audioFile } = this.state;
+
+    this.pathRef = firebase.storage().ref().child(audioURL);
```

Comment:
Should we be re-creating this pathRef on every mount? If audioFile already exists, doesn't that mean that the pathRef already existed? Maybe we should only initialize this pathRef when we don't have an audioFile?

Reply:
Thanks for pointing this out. Turns out I only used `pathRef` once to get the download URL. I've done away with `pathRef` completely.